Highbar.Types.Test/ValidationTests.cs

```diff
@@ -0,0 +1,1284 @@
+using System;
+using System.Collections.Generic;
+
+using Xunit;
+using Moq;
+using System.Linq;
+
+using static Highbar.Functions.Functions;
+
+namespace Highbar.Types.Tests
+{
+  public class ValidationTests
+  {
+    private static readonly string _testError = "testError";
+    private static readonly bool _testValue = true;
+
+    public class ValidationStatics
+    {
+      public class Attempt
+      {
+        [Fact]
+        public void ShouldReturnRightOfTheValue()
+        {
+          Func<bool> supplier = () => _testValue;
+          Validation<Exception, bool> expectedResult = Validation<Exception, bool>.Pass<Exception, bool>(_testValue);
+          Validation<Exception, bool> actualResult = Validation<Exception, bool>.Attempt(supplier);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnLeftOfTheException()
+        {
+          Func<bool> supplier = () => throw new NullReferenceException("value is null");
+          Validation<Exception, bool> expectedResult = Validation<Exception, bool>.Fail<Exception, bool>(new NullReferenceException("value is null"));
+          Validation<Exception, bool> actualResult = Validation<Exception, bool>.Attempt(supplier);
+
+          //NOTE(lee.crabtree):  The underlying exceptions only test for reference equality.  (/facepalm)
```

Comment:
This is likely because the stack trace between two exception are likely _not_ equivalent.